core-bundle/src/Image/Preview/PreviewProviderInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Image\Preview;
+
+use Contao\Image\ImageDimensions;
+
+interface PreviewProviderInterface
+{
+    public function supports(string $path, string $fileHeader = ''): bool;
+
+    /**
+     * @param int $size Minimal size the preview image should have in each
+     *                  dimension. Can be larger for implementation reasons or
+     *                  smaller if there is not enough data.
+     */
+    public function generatePreview(string $sourcePath, int $size, string $targetPath): void;
+
+    /**
+     * Calculate the exact dimensions the preview image will have when generated.
+     *
+     * @param int    $size       See self::generatePreview()
+     * @param string $fileHeader First X bytes of the source file
```

Comment:
What's `X bytes`, who defines how many bytes are required for the file header?

Reply:
Maybe loop over all providers and ask how many bytes they need at least to determine the file header and then take `max()`?

Reply:
> Maybe loop over all providers and ask how many bytes they need at least to determine the file header and then take `max()`?

👍 This way we can also not read the file at all if no provider needs it 💪

Reply:
1cd029830d63475ec9d74eca2c19945e0852ffd2